vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -306,6 +307,9 @@ public StructVector getStruct() {
 
   public ${name}Vector get${name}Vector(String name) {
     if (${uncappedName}Vector == null) {
+      ${uncappedName}Vector = internalStruct.getChild(fieldName(MinorType.${name?upper_case}), ${name}Vector.class);
+      if (${uncappedName}Vector == null) {
+        throw new IllegalArgumentException("No ${uncappedName} present. Provide ArrowType argument to create a new vector");
```

Comment:
Is it just me or is there no closing brace here?

Shouldn't we only add this new check for parameterized/nested types?

Reply:
From a syntax standpoint (brace), it looks good to me.

Reply:
I pushed a new commit where this code is moved slightly, had forgotten that this one was updated in our repo.